signature-specification.md

```diff
@@ -1,243 +1,118 @@
-The process is described below:
-
-1. Compute the Base64Url value of ProtectedHeaders.
-1. Compute the Base64Url value of JWSPayload.
-1. Build message to be signed by concatenating the values generated in step 1 and step 2 using '.'
-`ASCII(BASE64URL(UTF8(ProtectedHeaders)) ‘.’ BASE64URL(JWSPayload))`
-1. Compute the signature on the message constructed in the previous step by using the signature algorithm defined in the corresponding header element: `alg`.
-1. Compute the Base64Url value of the signature produced in the previous step.
-   This is the value of the signature property used in the signature envelope.
-
-**Signature Envelope**: The final signature envelope comprises of Claims, ProtectedHeaders, UnprotectedHeaders, and signature.
-
-Since Notary v2 restricts one signature per signature envelope, the compliant signature envelope MUST be in flattened JWS JSON format.
+### Unsigned Attributes
 
-```json
-{
-    "payload": "<Base64Url(JWSPayload)>",
-    "protected": "<Base64Url(ProtectedHeaders)>",
-    "header": {
-        "timestamp": "<Base64(TimeStampToken)>",
-        "x5c": ["<Base64(DER(leafCert))>", "<Base64(DER(intermediateCACert))>", "<Base64(DER(rootCert))>"]
-    },
-    "signature": "Base64Url( sign( ASCII( <Base64Url(ProtectedHeader)>.<Base64Url(JWSPayload)> )))"  
-}
-```
+These attributes are considered unsigned with respect to the signing key that generates the signature. These attributes are typically signed by a third party (e.g. CA, TSA).
 
-**Implementation Constraints**: Notary v2 implementation MUST enforce the following constraints on signature generation and verification:
-
-1. `alg` header value MUST NOT be `none` or any symmetric-key algorithm such as `HMAC`.
-1. `alg` header value MUST be same as that of signature algorithm identified using signing certificate's public key algorithm and size.
-1. `alg` header values for various signature algorithms:
-  | Signature Algorithm             | `alg` Param Value |
-  | ------------------------------- | ----------------- |
-  | RSASSA-PSS with SHA-256         | PS256             |
-  | RSASSA-PSS with SHA-384         | PS384             |
-  | RSASSA-PSS with SHA-512         | PS512             |
-  | ECDSA on secp256r1 with SHA-256 | ES256             |
-  | ECDSA on secp384r1 with SHA-384 | ES384             |
-  | ECDSA on secp521r1 with SHA-512 | ES512             |
-1. Signing certificate MUST be a valid codesigning certificate.
-1. Only JWS JSON flattened format is supported.
-   See 'Signature Envelope' section.
+- **Certificate Chain**: This property contains the list of X.509 certificate or certificate chain. This is a REQUIRED attribute. The certificate chain is authenticated using against a trust store as part of signature validation.
+- **TSA counter signature** : The time stamp token generated for a given signature. Only [RFC3161](ietf-rfc3161) compliant TimeStampToken are supported. This is an OPTIONAL attribute.
```

Comment:
Mention that TSA is a trusted time stamp.

Reply:
Are there any guidelines on how a signer should encode the TSA signature into an unsigned attribute?

Reply:
I'm assuming ASN.1 DER encoded, then base64 encoded. Definitely has to be spelled out.

Reply:
Nevermind, this is a serialization detail that has to go into the JWS doc.

Reply:
Yes, this detail should go in the specific envelope format doc.